dlp/api/DlpSample/Program.cs

```diff
@@ -0,0 +1,159 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using CommandLine;
+using Google.Cloud.Dlp.V2;
+using Google.Protobuf;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace GoogleCloudSamples
+{
+    abstract class InspectLocalOptions
+    {
+        [Value(0, HelpText = "The project ID to run the API call under.", Required = true)]
+        public string ProjectId { get; set; }
+        [Option('i', "info-types", HelpText = "Comma-separated infoTypes of information to match.",
+            Default = "PHONE_NUMBER,EMAIL_ADDRESS,CREDIT_CARD_NUMBER")]
+        public string InfoTypes { get; set; }
+        [Option('l', "minimum-likelihood",
+            HelpText = "The minimum likelihood required before returning a match (0-5).", Default = 0)]
+        public int MinLikelihood { get; set; }
+        [Option('m', "max-findings",
+            HelpText = "The maximum number of findings to report per request (0 = server maximum).", Default = 0)]
+        public int MaxFindings { get; set; }
+        [Option('n', "no-includeQuotes", HelpText = "Do not include matching quotes.")]
+        public bool NoIncludeQuote { get; set; }
+    }
+
+    [Verb("inspectString", HelpText = "Inspects a content string.")]
+    class InspectStringOptions : InspectLocalOptions
+    {
+        [Value(1, HelpText = "The item to inspect.", Required = true)]
+        public string Value { get; set; }
+    }
+
+    [Verb("inspectFile", HelpText = "Inspects a content file.")]
+    class InspectFileOptions : InspectLocalOptions
+    {
+        [Value(1, HelpText = "The path to the local file to inspect. Can be a text, JPG, or PNG.", Required = true)]
+        public string File { get; set; }
+    }
+
+    public class Dlp
+    {
+        static IEnumerable<InfoType> ParseInfoTypes(string infoTypesStr)
+        {
+            return infoTypesStr.Split(',').Select(str =>
+            {
+                try
+                {
+                    return InfoType.Parser.ParseJson($"{{\"name\": \"{str}\"}}");
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine($"Failed to parse infoType {str}: {e}");
+                    return null;
+                }
+            }).Where(it => it != null);
+        }
+
+        static object InspectString(InspectStringOptions opts)
+        {
+            return InspectLocal(opts, new InspectContentRequest
+            {
+                Parent = $"projects/{opts.ProjectId}",
+                Item = new ContentItem
+                {
+                    Value = opts.Value
+                }
+            });
+        }
+
+        static readonly Dictionary<string, ByteContentItem.Types.BytesType> fileTypes =
+            new Dictionary<string, ByteContentItem.Types.BytesType>()
+        {
+            {"bmp", ByteContentItem.Types.BytesType.ImageBmp},
```

Comment:
Are the string and enum's string value the same? If yes, worth changing keys to [ENUM].ToString()

Reply:
They're not the same. The keys are the actual extensions of file types, and the values are the BytesType enum values that all start with "Image" or have their encoding or something.